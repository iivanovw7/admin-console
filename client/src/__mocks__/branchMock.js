export const BRANCH = {
  address: '938 Barby Way',
  email: 'mbrazier0@amazon.co.uk',
  fax: '+86 829 425 7497',
  information: '1',
  name: 'San Mart√≠n de los Andes',
  phone: '+54 275 474 9318',
  status: true,
  __v: 0,
  _id: '5c8f96b9cece9911f01dd092'
};