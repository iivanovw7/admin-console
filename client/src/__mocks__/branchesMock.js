export const BRANCHES = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      address: '938 Barby Way',
      email: 'mbrazier0@amazon.co.uk',
      fax: '+86 829 425 7497',
      information: '1',
      name: 'San Mart√≠n de los Andes',
      phone: '+54 275 474 9318',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd092'
    },
    {
      address: '903 Dottie Street',
      email: 'inerger1@businesswire.com',
      fax: '+86 230 609 6364',
      information: 't',
      name: 'Dingzhai',
      phone: '+86 103 802 9655',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd093'
    },
    {
      status:false,
      _id:"5c8f96b9cece9911f01dd0a0",
      name:"Cantao-an",
      email:"atroye@dailymail.co.uk",
      phone:"+63 269 360 0515",
      fax:"+86 713 306 1827",
      address:"7827 Luster Point",
    }
  ]
};
