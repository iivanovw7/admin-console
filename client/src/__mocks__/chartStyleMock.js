export const style = {
  type: "PieChart",
  id: 2
};
