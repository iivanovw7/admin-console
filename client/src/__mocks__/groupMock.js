export const GROUP = {
  status: true,
  permissions: true,
  _id: '5c8f96bacece9911f01dd0a6',
  name: 'Front-End Pool',
  description: 'Group of all employees with Front-End expertise'
};