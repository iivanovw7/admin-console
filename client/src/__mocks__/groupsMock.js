export const GROUPS = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      status: true,
      permissions: true,
      _id: '5c8f96bacece9911f01dd0a6',
      name: 'Front-End Pool',
      description: 'Group of all employees with Front-End expertise',

    },
    {
      status: false,
      permissions: false,
      _id: '5c8f96bacece9911f01dd0a7',
      name: 'Back-End Pool',
      description: 'Group of all employees with Back-End expertise',
    },
    {
      status:false,
      permissions:false,
      _id:"5cade7a578f0a97097c972c5",
      name:"TEST",
      description:"TEST",
    }
  ]
};
