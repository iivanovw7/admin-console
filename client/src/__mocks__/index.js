export * from './branchesMock';
export * from './branchMock';
export * from './userMock';
export * from './usersMock';