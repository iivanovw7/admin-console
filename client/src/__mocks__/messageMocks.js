export const MESSAGE = {
  _id: '5cb83f6a5042891fe2afa161',
  subject: 'ssss',
  message: 'sssss',
  senderId: {
    status: true,
    _id: '5c8f97c5cece9911f01dd561',
    name: 'Admin',
    surname: 'ADMIN',
    email: 'admin@company.org',
    role: '5c8f96bbcece9911f01dd0ab',
    password: '$2a$10$mNoHh1swsVyvIu4IILOoCu5E8FKFynxXHEe5G3GpYA8OjCpGcs3l.',
    created: '2019-03-18T13:06:13.037Z',
    branch: '5c8f96b9cece9911f01dd0a0',
    group: '5cade7a578f0a97097c972c5'
  },
  branchId: {
    status: false,
    _id: '5c8f96b9cece9911f01dd0a0',
    name: 'Cantao-an',
    email: 'atroye@dailymail.co.uk',
    phone: '+63 269 360 0515',
    fax: '+86 713 306 1827',
    address: '7827 Luster Point'
  },
  created: '2019-04-18T09:12:10.783Z'
};
