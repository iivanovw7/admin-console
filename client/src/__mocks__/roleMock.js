export const ROLE = {
  public: false,
  editable: false,
  _id: '5c8f96bbcece9911f01dd0ab',
  name: 'Administrator',
  code: 'ADMIN',
  description: 'Main System Administrator with full access',
  active: true
};

export const CHANGED_ROLE = {
  newRole: {
    public: false,
    editable: false,
    _id: '5c8f96bbcece9911f01dd0ac',
    name: 'Branch Administrator',
    code: 'BRANCH_ADMIN',
    description: 'Administrator with full access to their own branch',
    active: true
  }
};

export const ROLE_USER = {
  public:false,
  editable:false,
  _id:"5c8f96bbcece2211f01dd0ab",
  name:"User",
  code:"USER",
  description:"Default user",
  active:true
};
