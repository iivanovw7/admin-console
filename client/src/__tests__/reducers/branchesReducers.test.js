import * as types from '../../constants/actionTypes';
import branchesReducer from '../../reducers/branchesReducer';
import * as mocks from './../../__mocks__/';

const setup = () => {

  describe('BRANCH reducers', () => {

    it('FETCH BRANCHES', () => {
      const expectedActions = {
        type: types.FETCH_BRANCHES,
        payload: {
          data: mocks.BRANCHES
        }
      };

      expect(branchesReducer({}, expectedActions)).toMatchSnapshot();
    });

    it('FETCH BRANCH', () => {
      const expectedActions = {
        type: types.FETCH_BRANCH,
        payload: {
          data: mocks.BRANCH
        }
      };

      expect(branchesReducer({}, expectedActions)).toMatchSnapshot();
    });

    it('UPDATE BRANCH', () => {
      const expectedActions = {
        type: types.UPDATE_BRANCH,
        payload: {
          data: mocks.BRANCH
        }
      };

      expect(branchesReducer({}, expectedActions)).toMatchSnapshot();
    });

    it('ADD BRANCH', () => {
      const expectedActions = {
        type: types.ADD_BRANCH,
        payload: {
          data: mocks.BRANCH
        }
      };

      expect(branchesReducer({}, expectedActions)).toMatchSnapshot();
    });

  });

};

setup();