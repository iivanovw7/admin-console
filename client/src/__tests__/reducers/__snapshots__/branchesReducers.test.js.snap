// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BRANCH reducers ADD BRANCH 1`] = `
Object {
  "branch": Object {},
  "confirmed": false,
  "error": null,
  "success": "Branch \\"San Martín de los Andes\\" created!",
}
`;

exports[`BRANCH reducers FETCH BRANCH 1`] = `
Object {
  "branch": Object {
    "__v": 0,
    "_id": "5c8f96b9cece9911f01dd092",
    "address": "938 Barby Way",
    "email": "mbrazier0@amazon.co.uk",
    "fax": "+86 829 425 7497",
    "information": "1",
    "name": "San Martín de los Andes",
    "phone": "+54 275 474 9318",
    "status": true,
  },
  "error": null,
  "success": null,
}
`;

exports[`BRANCH reducers FETCH BRANCHES 1`] = `
Object {
  "branch": Object {},
  "error": null,
  "list": Object {
    "limit": 7,
    "output": Array [
      Object {
        "__v": 0,
        "_id": "5c8f96b9cece9911f01dd092",
        "address": "938 Barby Way",
        "email": "mbrazier0@amazon.co.uk",
        "fax": "+86 829 425 7497",
        "information": "1",
        "name": "San Martín de los Andes",
        "phone": "+54 275 474 9318",
        "status": true,
      },
      Object {
        "__v": 0,
        "_id": "5c8f96b9cece9911f01dd093",
        "address": "903 Dottie Street",
        "email": "inerger1@businesswire.com",
        "fax": "+86 230 609 6364",
        "information": "t",
        "name": "Dingzhai",
        "phone": "+86 103 802 9655",
        "status": true,
      },
    ],
    "page": 1,
    "pages": 4,
    "results": 25,
  },
  "success": null,
}
`;

exports[`BRANCH reducers UPDATE BRANCH 1`] = `
Object {
  "branch": Object {},
  "confirmed": false,
  "error": null,
  "success": "Branch \\"San Martín de los Andes\\" successfully modified!",
}
`;
